UnixDomainServerConnector throws misleading exception on invalid socket path

**Jetty version(s)**
11.0.7

**Java version/vendor** `(use: java -version)`
```
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment (build 16+36-2231)
OpenJDK 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)
```

**OS type/version**
```
Distributor ID:	Linuxmint
Description:	Linux Mint 20.1
Release:	20.1
Codename:	ulyssa
```

**Description**
Whenever any `Throwable` is thrown binding to a UNIX domain socket, `UnixDomainServerConnector` wraps it in an `UnsupportedOperationException` with message `Unix-Domain SocketChannels are available starting from Java 16, your Java version is: <your java version>`.  However, a `Throwable` might be thrown for reasons other than the Java version being too early, for example if the path specified for the domain socket does not exist.

Two obvious solutions are:

- _either_ [the `catch` block](https://github.com/eclipse/jetty.project/blob/jetty-11.0.7/jetty-unixdomain-server/src/main/java/org/eclipse/jetty/unixdomain/server/UnixDomainServerConnector.java#L249) should catch a more specific type;
- _or_ [the wrapping exception and its message](https://github.com/eclipse/jetty.project/blob/jetty-11.0.7/jetty-unixdomain-server/src/main/java/org/eclipse/jetty/unixdomain/server/UnixDomainServerConnector.java#L251) should be more general

I'm not sure which is preferable.

**How to reproduce?**

```java
package com.example;

import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.unixdomain.server.UnixDomainServerConnector;

import java.nio.file.Path;

public class Example {
    public static void main(String... args) throws Exception {
        var server = new Server();
        var unixDomainServerConnector = new UnixDomainServerConnector(server);
        unixDomainServerConnector.setUnixDomainPath(Path.of("/does/not/exist"));
        server.addConnector(unixDomainServerConnector);
        server.start();
    }
}
```
Results in the following on `stderr`:
```
Exception in thread "main" java.lang.UnsupportedOperationException: Unix-Domain SocketChannels are available starting from Java 16, your Java version is: 16
	at org.eclipse.jetty.unixdomain.server@11.0.7/org.eclipse.jetty.unixdomain.server.UnixDomainServerConnector.bindServerSocketChannel(UnixDomainServerConnector.java:252)
	at org.eclipse.jetty.unixdomain.server@11.0.7/org.eclipse.jetty.unixdomain.server.UnixDomainServerConnector.openServerSocketChannel(UnixDomainServerConnector.java:232)
	at org.eclipse.jetty.unixdomain.server@11.0.7/org.eclipse.jetty.unixdomain.server.UnixDomainServerConnector.open(UnixDomainServerConnector.java:203)
	at org.eclipse.jetty.unixdomain.server@11.0.7/org.eclipse.jetty.unixdomain.server.UnixDomainServerConnector.doStart(UnixDomainServerConnector.java:151)
	at org.eclipse.jetty.util@11.0.7/org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
	at org.eclipse.jetty.server@11.0.7/org.eclipse.jetty.server.Server.doStart(Server.java:427)
	at org.eclipse.jetty.util@11.0.7/org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
	at com.example/com.example.Example.main(Example.java:14)
Caused by: java.net.SocketException: No such file or directory
Caused by: java.net.SocketException: No such file or directory

	at java.base/sun.nio.ch.UnixDomainSockets.bind0(Native Method)
	at java.base/sun.nio.ch.UnixDomainSockets.bind(UnixDomainSockets.java:111)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.unixBind(ServerSocketChannelImpl.java:319)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:292)
	at org.eclipse.jetty.unixdomain.server@11.0.7/org.eclipse.jetty.unixdomain.server.UnixDomainServerConnector.bindServerSocketChannel(UnixDomainServerConnector.java:246)
	... 7 more
```

--- a/jetty-unixdomain-server/src/main/java/org/eclipse/jetty/unixdomain/server/UnixDomainServerConnector.java
+++ b/jetty-unixdomain-server/src/main/java/org/eclipse/jetty/unixdomain/server/UnixDomainServerConnector.java
@@ -233,23 +233,36 @@ public class UnixDomainServerConnector extends AbstractConnector
         return serverChannel;
     }
 
-    private ServerSocketChannel bindServerSocketChannel()
+    private ServerSocketChannel bindServerSocketChannel() throws IOException
     {
+        Path unixDomainPath = getUnixDomainPath();
+
+        ServerSocketChannel serverChannel;
+        SocketAddress socketAddress;
         try
         {
             ProtocolFamily family = Enum.valueOf(StandardProtocolFamily.class, "UNIX");
             Class<?> channelClass = Class.forName("java.nio.channels.ServerSocketChannel");
-            ServerSocketChannel serverChannel = (ServerSocketChannel)channelClass.getMethod("open", ProtocolFamily.class).invoke(null, family);
+            serverChannel = (ServerSocketChannel)channelClass.getMethod("open", ProtocolFamily.class).invoke(null, family);
             // Unix-Domain does not support SO_REUSEADDR.
             Class<?> addressClass = Class.forName("java.net.UnixDomainSocketAddress");
-            SocketAddress socketAddress = (SocketAddress)addressClass.getMethod("of", Path.class).invoke(null, getUnixDomainPath());
-            serverChannel.bind(socketAddress, getAcceptQueueSize());
-            return serverChannel;
+            socketAddress = (SocketAddress)addressClass.getMethod("of", Path.class).invoke(null, unixDomainPath);
         }
         catch (Throwable x)
         {
             String message = "Unix-Domain SocketChannels are available starting from Java 16, your Java version is: " + JavaVersion.VERSION;
             throw new UnsupportedOperationException(message, x);
+        }
+
+        try
+        {
+            serverChannel.bind(socketAddress, getAcceptQueueSize());
+            return serverChannel;
+        }
+        catch (IOException x)
+        {
+            String message = String.format("Could not bind %s to %s", UnixDomainServerConnector.class.getSimpleName(), unixDomainPath);
+            throw new IOException(message, x);
         }
     }
 

--- a/jetty-unixdomain-server/src/test/java/org/eclipse/jetty/unixdomain/server/UnixDomainTest.java
+++ b/jetty-unixdomain-server/src/test/java/org/eclipse/jetty/unixdomain/server/UnixDomainTest.java
@@ -13,6 +13,7 @@
 
 package org.eclipse.jetty.unixdomain.server;
 
+import java.io.IOException;
 import java.net.SocketAddress;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -53,6 +54,7 @@ import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @EnabledForJreRange(min = JRE.JAVA_16)
@@ -254,6 +256,16 @@ public class UnixDomainTest
         {
             httpClient.stop();
         }
+    }
+
+    @Test
+    public void testInvalidUnixDomainPath()
+    {
+        server = new Server();
+        UnixDomainServerConnector connector = new UnixDomainServerConnector(server, factories);
+        connector.setUnixDomainPath(Path.of("/does/not/exist"));
+        server.addConnector(connector);
+        assertThrows(IOException.class, () -> server.start());
     }
 
     private static Path toUnixDomainPath(SocketAddress address)

